File: frontend/types/employee.ts
/**
 * 직원 관련 타입 정의
 */

// 직원 정보
export interface Employee {
  id: string;
  name: string;
  email?: string;
  phone?: string;
  employment_type: 'regular' | 'contract' | 'part_time' | 'temporary';
  position?: string;
  department?: string;
  start_date: string;
  end_date?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

// 직원 등록 요청
export interface CreateEmployeeRequest {
  name: string;
  email?: string;
  phone?: string;
  employment_type: 'regular' | 'contract' | 'part_time' | 'temporary';
  position?: string;
  department?: string;
  start_date: string;
  end_date?: string;
}

// 직원 수정 요청
export interface UpdateEmployeeRequest {
  name?: string;
  email?: string;
  phone?: string;
  employment_type?: 'regular' | 'contract' | 'part_time' | 'temporary';
  position?: string;
  department?: string;
  end_date?: string;
}

// 직원 목록 응답
export interface EmployeeListResponse {
  data: Employee[];
  pagination?: {
    cursor: string | null;
    hasNext: boolean;
    limit: number;
  };
}

File: backend/app/schemas/employee.py
# Employee 관련 스키마
from datetime import datetime, date
from typing import Literal
from pydantic import BaseModel, Field
import uuid


class EmployeeBase(BaseModel):
    """Employee 기본 스키마"""
    name: str = Field(..., min_length=1, max_length=100)
    nationality: Literal["korean", "chinese", "vietnamese", "american", "other"] = "korean"
    employment_type: Literal["regular", "fixed_term", "part_time", "daily", "dispatch", "probation"]
    department: str | None = None
    position: str | None = None
    hire_date: date
    phone: str | None = None
    email: str | None = None
    bank_name: str | None = None
    bank_account: str | None = None


class EmployeeCreate(EmployeeBase):
    """Employee 생성"""
    id_number: str | None = None


class EmployeeUpdate(BaseModel):
    """Employee 수정"""
    name: str | None = Field(None, min_length=1, max_length=100)
    department: str | None = None
    position: str | None = None
    phone: str | None = None
    email: str | None = None
    bank_name: str | None = None
    bank_account: str | None = None
    is_active: bool | None = None


class EmployeeResponse(EmployeeBase):
    """Employee 응답"""
    id: uuid.UUID
    company_id: uuid.UUID
    user_id: uuid.UUID | None
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class EmployeeListResponse(BaseModel):
    """Employee 목록 응답 (간단)"""
    id: uuid.UUID
    name: str
    employment_type: str
    is_active: bool
    hire_date: date
